docs/content/commands/npm-pkg.md

```diff
@@ -0,0 +1,117 @@
+---
+title: npm-pkg
+section: 1
+description: Manages your package.json
+---
+
+### Synopsis
+
+```bash
+npm pkg set <key>=<value> [<key>=<value> ...]
+npm pkg get [<key> [<key> ...]]
+npm pkg delete <key> [<key> ...]
+```
+
+### Description
+
+A command that automates the management of `package.json` files.
+
+* `npm get <key>`
+
+    Retrieves a value `key` defined in your `package.json` file.
+
+* `npm set <key>=<value>`
+
+    Sets a `value` in your `package.json` based on the `key`
+
+* `npm delete <key>`
+
+    Deletes a `key` from your `package.json`
+
+### Workspaces support
+
+You can set/get/delete items across your configured workspaces by using the
+`workspace` or `workspaces` config options.
+
+For example, setting `funding` value across all configured workspaces
+of a project:
+
+```
+npm pkg set funding=https://example.com --ws
+```
+
+### Configuration
+
+<!-- AUTOGENERATED CONFIG DESCRIPTIONS START -->
+<!-- automatically generated, do not edit manually -->
+#### `force`
+
+* Default: false
+* Type: Boolean
+
+Removes various protections against unfortunate side effects, common
+mistakes, unnecessary performance degradation, and malicious input.
+
+* Allow clobbering non-npm files in global installs.
+* Allow the `npm version` command to work on an unclean git repository.
+* Allow deleting the cache folder with `npm cache clean`.
+* Allow installing packages that have an `engines` declaration requiring a
+  different version of npm.
+* Allow installing packages that have an `engines` declaration requiring a
+  different version of `node`, even if `--engine-strict` is enabled.
+* Allow `npm audit fix` to install modules outside your stated dependency
+  range (including SemVer-major changes).
+* Allow unpublishing all versions of a published package.
+* Allow conflicting peerDependencies to be installed in the root project.
+* Implicitly set `--yes` during `npm init`.
+
+If you don't have a clear idea of what you want to do, it is strongly
+recommended that you do not use this option!
+
+#### `pkg-type`
```

Comment:
yeah, that's slightly better but I'd prefer something shorter

Reply:
hmm - instead of this approach (tell me what exact type to coerce to), why not have the default be "string mode", and a single boolean config (`--json`) that means "use JSON syntax"?

iow, `npm pkg set 'foo=bar'` would be equivalent to `npm pkg set --json 'foo="bar"'`, and to set numbers or booleans, i'd just omit the quotes.

Reply:
hmmm I'm really not sure I like that idea, having to use quotes most of the time (most of `package.json` properties are strings anyways) is very annoying an it's going to not be very intuitive.

what about a shorter name, e.g: `--pkg-cast` that defaults to `string` and possible values are `number`|`boolean`?

Reply:
maybe i wasn't being clear - the common case would be "setting a string, do not use --json or quotes" and "setting a boolean or a number or null, use --json and no quotes".

Reply:
I'm having a hard time visualizing it, could you make some examples showing how to:
- setting `"private": true` in a package.json
- setting `"tap": { timeout: 60 }` in a package.json

with a coercion key-based prop, it'd be something like:
```bash
$ npm pkg set private=true --pkg-cast=boolean
$ npm pkg set tap.timeout=60 --pkg-cast=number
```

Reply:
```sh
$ npm pkg set --json private=true
$ npm pkg set --json tap.timeout=60
```

Reply:
oohh I see, so basically `--json` means the only type casting we're doing is to run `JSON.parse` on the value we take from the provided `key=value` otherwise we default to the original string value... 🤔 did I get it right this time?

Reply:
it does sounds much cleaner from a syntax point of view, although I'm afraid this might lead users to think objects or arrays are also supported, which is not the case as of today

Reply:
Sure - but we could throw on that, and they'd learn immediately :-)

it also lends itself to trivial syntax for objects and arrays if we want to add it later.

Reply:
cool 👍 let's do it! 😄

Reply:
(i'm pretty sure from @isaacs saying "--json is for the output format" in the RFC call, that we'll want to bikeshed the spelling of the flag, but this does define the output format, just _also_ part of the input format :-p)

Reply:
yep, I'll leave it as `--json` for now and we may tweak it prior to landing the PR if needed 😊